.github/workflows/rust.yml

```
@@ -47,7 +48,15 @@ jobs:
     - run: rustup update
     - run: cargo version
     - run: cargo build --target ${{ matrix.target }}
-    - run: cargo test --target ${{ matrix.target }}
+    - name: Run tests with std
+      if: matrix.features == 'std'
+      run: cargo test --target ${{ matrix.target }} --features std
+    - name: Run tests with alloc
+      if: matrix.features == 'alloc'
+      run: cargo test --target ${{ matrix.target }} --features alloc
+    - name: Run tests with no_alloc
+      if: matrix.features == 'no_alloc'
+      run: cargo test --target ${{ matrix.target }} --no-default-features
```

On other repos where we want to test a variety of features, we typically use cargo-hack. It does a good job of testing feature combinations. It'll test the default case too.
```suggestion
    - uses: stellar/binaries@v30
      with:
        name: cargo-hack
        version: 0.5.28
    - run: cargo-hack hack --feature-powerset --target ${{ matrix.sys.target }}
```